ARDemos/ARDemos/BottomBar.swift

```diff
@@ -0,0 +1,62 @@
+import UIKit
+
+class BottomBar: UIView {
+
+    private let scrollView = UIScrollView()
+    private var models: [Model]?
+    
+    var onTap: ((String) -> Void)?
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setUpViews()
+        setUpLayout()
+    }
+
+    private func setUpViews() {
+        addSubview(scrollView)
+    }
+
+    private func setUpLayout() {
+        applyConstraint(for: .top)
+        applyConstraint(for: .right)
+        applyConstraint(for: .bottom)
+        applyConstraint(for: .left)
+    }
+    
+    private func applyConstraint(for attribute: NSLayoutAttribute) {
+        let constraint = NSLayoutConstraint(item: scrollView,
+                                  attribute: attribute,
+                                  relatedBy: .equal,
+                                  toItem: self,
+                                  attribute: attribute,
+                                  multiplier: 1,
+                                  constant: 0)
+        
+        scrollView.addConstraint(constraint)
+    }
+
+    func addModelButtons(models: [Model]) {
```

Comment:
This method would be way safer if you call it `setup(with models: [Model])` 
and you remove any eventual subviews before to create the new ones. 

In this way you are safe to call this methods every-time you want.

Reply:
When you say "remove any eventual subviews before to create the new ones" do you mean, check if any subviews exist in the stackView and consequently remove them?